datashuttle/tui/app.py

```diff
@@ -0,0 +1,72 @@
+from pathlib import Path
+from time import monotonic
+
+from textual.app import App, ComposeResult
+from textual.binding import Binding
+from textual.widgets import (
+    DirectoryTree,
+    Footer,
+    Header,
+    Input,
+    Label,
+    TabbedContent,
+    TabPane,
+)
+
+
+class TuiApp(App):
+    """
+    The main app page for the DataShuttle TUI.
+
+    Running this application in a main block as below
+    if __name__ == __main__:
+         app = MyApp()
+         app.run()
+
+    Initialises the TUI event loop and starts the application.
+    """
+
+    TITLE = "DataShuttle"
+
+    # Set path to TCSS
+    tui_path = Path(__file__).parent
+    CSS_PATH = tui_path / "css" / "tab_content.tcss"
+
+    # Set key-bindings
+    BINDINGS = [
+        Binding("ctrl+d", "toggle_dark", "Toggle Dark Mode", priority=True)
+    ]
+
+    # Compose window
```

Comment:
I would suggest moving this to a docstring on the `compose` function and expanding a little bit as to what this is doing i.e. I think composing the widgets (in the order specified) onto the TUI).